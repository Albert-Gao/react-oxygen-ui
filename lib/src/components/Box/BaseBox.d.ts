import React from 'react';
import { IBaseBox } from './BaseBox.type';
export declare const BaseBox: React.FunctionComponent<IBaseBox>;
