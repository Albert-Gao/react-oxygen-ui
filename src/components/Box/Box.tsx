import * as React from "react";
import { getRefProp, is } from "../../utils/utils";
import { BaseBox } from "./BaseBox";
import { IBaseBoxWithTakeRef, PlainObject, Responsive } from "./BaseBox.type";
import { StyledGapBox } from "./StyledGapBox";

interface IA11Y {
  a11yTitle?: string;
}

export interface IBox extends IBaseBoxWithTakeRef, IA11Y {
  fill?: "horizontal" | "vertical" | true | false;
  gap?: Responsive;
}

export const Box: React.FC<IBox> = ({
  fill,
  gap,
  direction,
  a11yTitle,
  children,
  takeRef,
  ...rest
}) => {
  let fillProps: PlainObject = {};
  if (fill === true) {
    fillProps = {
      height: "100%",
      width: "100%"
    };
  } else if (fill === "horizontal") {
    fillProps.width = "100%";
  } else if (fill === "vertical") {
    fillProps.height = "100%";
  }

  let ariaTitle: PlainObject = {};
  if (a11yTitle) {
    ariaTitle = {
      "aria-label": a11yTitle
    };
  }

  let cursor: PlainObject = {};
  if (is.function(rest.onClick)) {
    cursor = {
      cursor: "pointer",
      role: "button"

      // After we figure out how to remove that ugly outline onFocus
      // We can enable the following accessible props
      // tabIndex: '0',
      // onKeyDown: rest.onClick,
    };
  }

  const contents: React.ReactNode[] = [];
  if (gap) {
    let firstIndex: number;
    React.Children.forEach(children, (child, index) => {
      if (child) {
        if (firstIndex === undefined) {
          firstIndex = index;
        } else {
          contents.push(
            <StyledGapBox
              key={index}
              gap={gap}
              direction={direction || "row"}
            />
          );
        }
      }
      contents.push(child);
    });
  }

  return (
    <BaseBox
      display="flex"
      {...ariaTitle}
      {...getRefProp(takeRef)}
      {...rest}
      {...fillProps}
      {...cursor}
      direction={direction}
    >
      {contents.length > 0 ? contents : children}
    </BaseBox>
  );
};
