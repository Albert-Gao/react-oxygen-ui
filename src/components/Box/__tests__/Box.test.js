import 'jest-styled-components';
import 'jest-dom/extend-expect';
import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { Box } from '../Box';
import {
  DEFAULT_SPACE,
  defaultTheme,
  ThemeProvider,
  DEFAULT_FONT_SIZE,
} from '../../../styles';

const getMediaQuery = num => `screen and (min-width:${num}px)`;

afterEach(cleanup);

describe('Test <Box />: justify-content', () => {
  test('should convert start to flex-start', () => {
    const { firstChild } = render(
      <Box justifyContent="start" />,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule(
      'justify-content',
      'flex-start',
    );
  });

  test('should convert end to flex-end', () => {
    const { firstChild } = render(
      <Box justifyContent="end" />,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('justify-content', 'flex-end');
  });

  test('should convert between to space-between', () => {
    const { firstChild } = render(
      <Box justifyContent="between" />,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule(
      'justify-content',
      'space-between',
    );
  });

  test('should convert around to space-around', () => {
    const { firstChild } = render(
      <Box justifyContent="around" />,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule(
      'justify-content',
      'space-around',
    );
  });
});

describe('Test <Box />: align-items', () => {
  test('should convert start to flex-start', () => {
    const { firstChild } = render(
      <Box alignItems="start" />,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('align-items', 'flex-start');
  });

  test('should convert end to flex-end', () => {
    const { firstChild } = render(<Box alignItems="end" />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('align-items', 'flex-end');
  });
});

describe('Test <Box />: align-self', () => {
  test('should convert start to flex-start', () => {
    const { firstChild } = render(
      <Box alignSelf="start" />,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('align-self', 'flex-start');
  });

  test('should convert end to flex-end', () => {
    const { firstChild } = render(<Box alignSelf="end" />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('align-self', 'flex-end');
  });
});

describe('Test <Box />: flex-direction', () => {
  test('should accept row', () => {
    const { firstChild } = render(<Box direction="row" />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('flex-direction', 'row');
  });

  test('should accept column', () => {
    const { firstChild } = render(
      <Box direction="column" />,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('flex-direction', 'column');
  });
});

describe('Test <Box />: fill', () => {
  test('should support fill="horizontal"', () => {
    const { firstChild } = render(
      <Box fill="horizontal" />,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).not.toHaveStyleRule('height', '100%');
    expect(firstChild).toHaveStyleRule('width', '100%');
  });

  test('should support fill="vertical"', () => {
    const { firstChild } = render(<Box fill="vertical" />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('height', '100%');
    expect(firstChild).not.toHaveStyleRule('width', '100%');
  });

  test('should support fill=true', () => {
    const { firstChild } = render(<Box fill />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('height', '100%');
    expect(firstChild).toHaveStyleRule('width', '100%');
  });

  test('should support fill=false', () => {
    const { firstChild } = render(<Box fill={false} />).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).not.toHaveStyleRule('height', '100%');
    expect(firstChild).not.toHaveStyleRule('width', '100%');
  });
});

describe('Test <Box /> gap', () => {
  test('[no gap with children] it should not add gap between children', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box direction="row">
          <p>a</p>
          <p>b</p>
        </Box>
      </ThemeProvider>,
    ).container;
    // firstChild is <Box>
    const a = firstChild.children[0];
    const b = firstChild.children[1];

    expect(firstChild).toMatchSnapshot();

    // a + b = 2
    expect(firstChild.children.length).toEqual(2);
    expect(a.innerHTML).toEqual('a');
    expect(b.innerHTML).toEqual('b');
  });

  test('[direction=row] it should add gap between children', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box direction="row" gap={DEFAULT_SPACE.medium}>
          <p>a</p>
          <p>b</p>
        </Box>
      </ThemeProvider>,
    ).container;
    // firstChild is <Box>
    const a = firstChild.children[0];
    const gapA = firstChild.children[1];
    const b = firstChild.children[2];

    expect(firstChild).toMatchSnapshot();

    // a + gapA + b = 3
    expect(firstChild.children.length).toEqual(3);
    expect(a.innerHTML).toEqual('a');
    expect(gapA.tagName).toEqual('DIV');
    expect(gapA).toHaveStyleRule(
      'width',
      `${defaultTheme.space[DEFAULT_SPACE.medium]}px`,
    );
    expect(b.innerHTML).toEqual('b');
  });

  test('[direction=undefined] it should add gap between children', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box gap={DEFAULT_SPACE.xxSmall}>
          <p>a</p>
          <p>b</p>
        </Box>
      </ThemeProvider>,
    ).container;
    // firstChild is <Box>
    const a = firstChild.children[0];
    const gapA = firstChild.children[1];
    const b = firstChild.children[2];

    expect(firstChild).toMatchSnapshot();

    // a + gapA + b = 3
    expect(firstChild.children.length).toEqual(3);
    expect(a.innerHTML).toEqual('a');
    expect(gapA.tagName).toEqual('DIV');
    expect(gapA).toHaveStyleRule(
      'width',
      `${defaultTheme.space[DEFAULT_SPACE.xxSmall]}px`,
    );
    expect(b.innerHTML).toEqual('b');
  });

  test('[direction=column] it should add gap between children', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box direction="column" gap={DEFAULT_SPACE.xxLarge}>
          <p>a</p>
          <p>b</p>
        </Box>
      </ThemeProvider>,
    ).container;
    // firstChild is <Box>
    const a = firstChild.children[0];
    const gapA = firstChild.children[1];
    const b = firstChild.children[2];

    expect(firstChild).toMatchSnapshot();

    // a + gapA + b = 3
    expect(firstChild.children.length).toEqual(3);
    expect(a.innerHTML).toEqual('a');
    expect(gapA.tagName).toEqual('DIV');
    expect(gapA).toHaveStyleRule(
      'height',
      `${defaultTheme.space[DEFAULT_SPACE.xxLarge]}px`,
    );
    expect(b.innerHTML).toEqual('b');
  });

  test('[Multiple children] it should add gap between children', () => {
    const expectedLength = `${
      defaultTheme.space[DEFAULT_SPACE.xLarge]
    }px`;
    const { firstChild } = render(
      <ThemeProvider>
        <Box
          direction={['row', 'column', 'row']}
          gap={DEFAULT_SPACE.xLarge}
        >
          <p>a</p>
          <p>b</p>
          <p>c</p>
        </Box>
      </ThemeProvider>,
    ).container;
    // firstChild is <Box>
    const a = firstChild.children[0];
    const gapA = firstChild.children[1];
    const b = firstChild.children[2];
    const gapB = firstChild.children[3];
    const c = firstChild.children[4];

    expect(firstChild).toMatchSnapshot();

    // a + gapA + b + gapB + c = 5
    expect(firstChild.children.length).toEqual(5);
    expect(a.innerHTML).toEqual('a');
    expect(gapA.tagName).toEqual('DIV');
    expect(gapA).toHaveStyleRule('width', expectedLength);
    expect(gapA).toHaveStyleRule('width', expectedLength, {
      media: getMediaQuery(defaultTheme.breakpoints[0]),
    });
    expect(gapA).toHaveStyleRule('height', expectedLength, {
      media: getMediaQuery(defaultTheme.breakpoints[1]),
    });
    expect(b.innerHTML).toEqual('b');
    expect(gapB.tagName).toEqual('DIV');
    expect(gapB).toHaveStyleRule('width', expectedLength);
    expect(gapB).toHaveStyleRule('width', expectedLength, {
      media: getMediaQuery(defaultTheme.breakpoints[0]),
    });
    expect(gapB).toHaveStyleRule('height', expectedLength, {
      media: getMediaQuery(defaultTheme.breakpoints[1]),
    });
    expect(c.innerHTML).toEqual('c');
  });
});

describe('Test <Box />: multiple props', () => {
  test('should accept boxShadow', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box boxShadow={5} />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule(
      'box-shadow',
      defaultTheme.shadows[5],
    );
  });

  test('should accept fontSize', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box fontSize={DEFAULT_FONT_SIZE.body} />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule(
      'font-size',
      defaultTheme.fontSpecs[DEFAULT_FONT_SIZE.body].size,
    );
  });

  test('should accept mx', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box mx={DEFAULT_SPACE.small} />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule(
      'margin-left',
      defaultTheme.space[DEFAULT_SPACE.small] + 'px',
    );
    expect(firstChild).toHaveStyleRule(
      'margin-right',
      defaultTheme.space[DEFAULT_SPACE.small] + 'px',
    );
  });

  test('should accept py', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box py={DEFAULT_SPACE.large} />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule(
      'padding-top',
      defaultTheme.space[DEFAULT_SPACE.large] + 'px',
    );
    expect(firstChild).toHaveStyleRule(
      'padding-bottom',
      defaultTheme.space[DEFAULT_SPACE.large] + 'px',
    );
  });

  test('should accept p[]', () => {
    const getSpace = key =>
      defaultTheme.space[DEFAULT_SPACE[key]] + 'px';
    const { firstChild } = render(
      <ThemeProvider>
        <Box
          px={[
            DEFAULT_SPACE.small,
            DEFAULT_SPACE.large,
            DEFAULT_SPACE.xLarge,
          ]}
        />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();

    expect(firstChild).toHaveStyleRule(
      'padding-left',
      getSpace('large'),
      {
        media: getMediaQuery(defaultTheme.breakpoints[0]),
      },
    );

    expect(firstChild).toHaveStyleRule(
      'padding-right',
      getSpace('large'),
      {
        media: getMediaQuery(defaultTheme.breakpoints[0]),
      },
    );

    expect(firstChild).toHaveStyleRule(
      'padding-left',
      getSpace('xLarge'),
      {
        media: getMediaQuery(defaultTheme.breakpoints[1]),
      },
    );

    expect(firstChild).toHaveStyleRule(
      'padding-right',
      getSpace('xLarge'),
      {
        media: getMediaQuery(defaultTheme.breakpoints[1]),
      },
    );

    expect(firstChild).toHaveStyleRule(
      'padding-left',
      defaultTheme.space[DEFAULT_SPACE['small']] + 'px',
    );
    expect(firstChild).toHaveStyleRule(
      'padding-right',
      defaultTheme.space[DEFAULT_SPACE['small']] + 'px',
    );
  });

  test('should accept py[]', () => {
    const getSpace = key =>
      defaultTheme.space[DEFAULT_SPACE[key]] + 'px';
    const { firstChild } = render(
      <ThemeProvider>
        <Box
          py={[
            DEFAULT_SPACE.small,
            DEFAULT_SPACE.large,
            DEFAULT_SPACE.xLarge,
          ]}
        />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();

    expect(firstChild).toHaveStyleRule(
      'padding-top',
      getSpace('large'),
      {
        media: getMediaQuery(defaultTheme.breakpoints[0]),
      },
    );

    expect(firstChild).toHaveStyleRule(
      'padding-bottom',
      getSpace('large'),
      {
        media: getMediaQuery(defaultTheme.breakpoints[0]),
      },
    );

    expect(firstChild).toHaveStyleRule(
      'padding-top',
      getSpace('xLarge'),
      {
        media: getMediaQuery(defaultTheme.breakpoints[1]),
      },
    );

    expect(firstChild).toHaveStyleRule(
      'padding-bottom',
      getSpace('xLarge'),
      {
        media: getMediaQuery(defaultTheme.breakpoints[1]),
      },
    );

    expect(firstChild).toHaveStyleRule(
      'padding-top',
      defaultTheme.space[DEFAULT_SPACE['small']] + 'px',
    );
    expect(firstChild).toHaveStyleRule(
      'padding-bottom',
      defaultTheme.space[DEFAULT_SPACE['small']] + 'px',
    );
  });

  test('should accept a11yTitle', () => {
    const a11attr = 'aria-label';
    const a11text = 'accessibility test';

    const { firstChild } = render(
      <ThemeProvider>
        <Box a11yTitle={a11text} />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveAttribute(a11attr);
    expect(firstChild.getAttribute(a11attr)).toEqual(a11text);
  });
});

describe('Test <Box css={}/>: ', () => {
  test('css prop should be added', () => {
    const { firstChild } = render(
      <ThemeProvider>
        <Box css={'background-color: black;'} />
      </ThemeProvider>,
    ).container;
    expect(firstChild).toMatchSnapshot();
    expect(firstChild).toHaveStyleRule('background-color', 'black');
  });
});
