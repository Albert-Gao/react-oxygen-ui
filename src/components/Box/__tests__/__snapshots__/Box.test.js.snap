// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Box /> gap [Multiple children] it should add gap between children 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 96px;
}

@media screen and (min-width:768px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    width: 96px;
  }
}

@media screen and (min-width:1024px) {
  .c1 {
    height: 96px;
  }
}

<div
  class="c0"
  direction="row,column,row"
  display="flex"
>
  <p>
    a
  </p>
  <div
    class="c1"
    direction="row,column,row"
  />
  <p>
    b
  </p>
  <div
    class="c1"
    direction="row,column,row"
  />
  <p>
    c
  </p>
</div>
`;

exports[`Test <Box /> gap [direction=column] it should add gap between children 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 128px;
}

<div
  class="c0"
  direction="column"
  display="flex"
>
  <p>
    a
  </p>
  <div
    class="c1"
    direction="column"
  />
  <p>
    b
  </p>
</div>
`;

exports[`Test <Box /> gap [direction=row] it should add gap between children 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 48px;
}

<div
  class="c0"
  direction="row"
  display="flex"
>
  <p>
    a
  </p>
  <div
    class="c1"
    direction="row"
  />
  <p>
    b
  </p>
</div>
`;

exports[`Test <Box /> gap [direction=undefined] it should add gap between children 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

.c1 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 16px;
}

<div
  class="c0"
  display="flex"
>
  <p>
    a
  </p>
  <div
    class="c1"
    direction="row"
  />
  <p>
    b
  </p>
</div>
`;

exports[`Test <Box /> gap [no gap with children] it should not add gap between children 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  direction="row"
  display="flex"
>
  <p>
    a
  </p>
  <p>
    b
  </p>
</div>
`;

exports[`Test <Box />: align-items should convert end to flex-end 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: align-items should convert start to flex-start 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: align-self should convert end to flex-end 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: align-self should convert start to flex-start 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: fill should support fill="horizontal" 1`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
  width="100%"
/>
`;

exports[`Test <Box />: fill should support fill="vertical" 1`] = `
.c0 {
  box-sizing: border-box;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
  height="100%"
/>
`;

exports[`Test <Box />: fill should support fill=false 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: fill should support fill=true 1`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
  height="100%"
  width="100%"
/>
`;

exports[`Test <Box />: flex-direction should accept column 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  direction="column"
  display="flex"
/>
`;

exports[`Test <Box />: flex-direction should accept row 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  direction="row"
  display="flex"
/>
`;

exports[`Test <Box />: justify-content should convert around to space-around 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: justify-content should convert between to space-between 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: justify-content should convert end to flex-end 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: justify-content should convert start to flex-start 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: multiple props should accept a11yTitle 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  aria-label="accessibility test"
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: multiple props should accept boxShadow 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: 0px 1px 2px rgba(0,0,0,0.20);
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: multiple props should accept fontSize 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 14px;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
  font-size="1"
/>
`;

exports[`Test <Box />: multiple props should accept mx 1`] = `
.c0 {
  box-sizing: border-box;
  margin-left: 32px;
  margin-right: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: multiple props should accept p[] 1`] = `
.c0 {
  box-sizing: border-box;
  padding-left: 32px;
  padding-right: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

@media screen and (min-width:768px) {
  .c0 {
    padding-left: 64px;
    padding-right: 64px;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    padding-left: 96px;
    padding-right: 96px;
  }
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: multiple props should accept py 1`] = `
.c0 {
  box-sizing: border-box;
  padding-top: 64px;
  padding-bottom: 64px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box />: multiple props should accept py[] 1`] = `
.c0 {
  box-sizing: border-box;
  padding-top: 32px;
  padding-bottom: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
}

@media screen and (min-width:768px) {
  .c0 {
    padding-top: 64px;
    padding-bottom: 64px;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    padding-top: 96px;
    padding-bottom: 96px;
  }
}

<div
  class="c0"
  display="flex"
/>
`;

exports[`Test <Box css={}/>:  css prop should be added 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  transition: all 0.3s cubic-bezier(0.53,0.01,0.36,1.63);
  background-color: black;
}

<div
  class="c0"
  display="flex"
/>
`;
